tools/apprunner-common/src/main/java/org/projectnessie/quarkus/runner/InputBuffer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.quarkus.runner;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.function.Consumer;
+
+/** Captures input from an {@link InputStream} and emits full lines terminated with a {@code LF}. */
+final class InputBuffer {
```

Comment:
Unit tests for this? I can't read this to know its right. Is there really nothing in the standard library to read inputs and split by line breaks?